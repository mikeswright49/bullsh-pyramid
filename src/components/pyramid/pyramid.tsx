import { Card } from 'src/components/card/card';
import React from 'react';
import { Card as CardType } from 'types/card';

export function Pyramid({ tiers }: { tiers: CardType[][] }) {
    if (!tiers) {
        return null;
    }

    const renderedTiers: CardType[][] = [].concat(tiers).reverse();

    return (
        <div data-testid="at-pyramid">
            {renderedTiers.map((tier, tierIndex) => (
                <div key={`pyramid-tier-${tierIndex}`} className={'flex-center-center stack-y-3'}>
                    {tier.map((card: CardType, cardIndex) => (
                        <span
                            className="stack-x-4"
                            key={`pyramid-tier-${tierIndex}-card-${cardIndex}`}
                        >
                            <Card card={card} />
                        </span>
                    ))}
                </div>
            ))}
        </div>
    );
}
