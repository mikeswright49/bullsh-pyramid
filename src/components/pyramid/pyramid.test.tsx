import { render } from '@testing-library/react';
import React from 'react';
import { Pyramid } from './pyramid';
jest.mock('../../utilities/shortid');

describe('<Unit Test> Pyramid', () => {
    it('should be able to render the component', () => {
        const { container } = render(
            <Pyramid
                tiers={[
                    [
                        {
                            value: 1,
                            hidden: false,
                            suit: 0,
                        },
                        {
                            value: 2,
                            hidden: false,
                            suit: 0,
                        },
                        {
                            value: 3,
                            hidden: false,
                            suit: 0,
                        },
                    ],
                    [
                        {
                            value: 4,
                            hidden: false,
                            suit: 0,
                        },
                        {
                            value: 5,
                            hidden: false,
                            suit: 0,
                        },
                    ],
                    [
                        {
                            value: 6,
                            hidden: false,
                            suit: 0,
                        },
                    ],
                ]}
            />
        );
        expect(container).toMatchSnapshot();
    });
});
