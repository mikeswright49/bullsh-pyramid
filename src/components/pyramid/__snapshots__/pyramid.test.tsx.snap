// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Unit Test> Pyramid should be able to render the component 1`] = `
<div>
  <div
    data-testid="at-pyramid"
  >
    <div
      class="flex-center-center stack-y-3"
    >
      <div
        class="flex-center-center card"
        data-testid="card-0-6"
        style="background-position: -474px -0px;"
      />
    </div>
    <div
      class="flex-center-center stack-y-3"
    >
      <div
        class="flex-center-center card"
        data-testid="card-0-4"
        style="background-position: -316px -0px;"
      />
      <div
        class="flex-center-center card"
        data-testid="card-0-5"
        style="background-position: -395px -0px;"
      />
    </div>
    <div
      class="flex-center-center stack-y-3"
    >
      <div
        class="flex-center-center card"
        data-testid="card-0-1"
        style="background-position: -79px -0px;"
      />
      <div
        class="flex-center-center card"
        data-testid="card-0-2"
        style="background-position: -158px -0px;"
      />
      <div
        class="flex-center-center card"
        data-testid="card-0-3"
        style="background-position: -237px -0px;"
      />
    </div>
  </div>
</div>
`;
